src/java.base/share/classes/jdk/internal/jimage/ImageReaderFactory.java

```diff
@@ -46,9 +48,8 @@
 public class ImageReaderFactory {
     private ImageReaderFactory() {}
 
-    private static final String JAVA_HOME = System.getProperty("java.home");
     private static final Path BOOT_MODULES_JIMAGE =
-        Paths.get(JAVA_HOME, "lib", "modules");
+        Paths.get(StaticProperty.javaHome(),"lib", "modules");
```

Comment:
The jimage and jrtfs files can't use internal APIs as they are compiled to --release 8 for jrt-fs.jar.